service/test_automatic_run_experiment.py

```diff
@@ -82,3 +112,84 @@ def sort_key(dictionary):
 
     mocked_stop_experiment.assert_called_with(expected_experiment_name,
                                               expected_config_file)
+
+
+@pytest.mark.parametrize(
+    ('name', 'expected_result'), [('02000-1-1', False), ('2020-1-1', False),
+                                  ('2020-01-01', True),
+                                  ('2020-01-01-aflplusplus', True),
+                                  ('2020-01-01-1', True)])
+def test_validate_experiment_name(name, expected_result):
+    """Tests that validate experiment name returns True for valid names and
+    False for names that are not valid."""
+    assert (automatic_run_experiment.validate_experiment_name(name) ==
+            expected_result)
+
+
+# Call the parameter exp_request instead of request because pytest reserves it.
+@pytest.mark.parametrize(
+    ('exp_request', 'expected_result'),
+    [
+        ({
+            'experiment': EXPERIMENT,
+            'fuzzers': ['afl']
+        }, True),
+        (1, False),  # Not a dict.
+        ({
+            'experiment': EXPERIMENT,
+            'fuzzers': []
+        }, False),  # No fuzzers.
+        ({
+            'experiment': EXPERIMENT
+        }, False),  # No fuzzers.
+        ({
+            'fuzzers': ['afl']
+        }, False),  # No experiment.
+        # Invalid experiment.
```

Comment:
either move comments to end to match others, or line before block is better.

Reply:
Done.